src/sql/azdata.proposed.d.ts

```diff
@@ -297,6 +297,11 @@ declare module 'azdata' {
 		azureAccount?: string;
 		azureResourceId?: string;
 		azurePortalEndpoint?: string;
+
+		/**
+		 * Gets a boolean value of if the user specified a database to connect to
+		 */
+		connectToDatabase?: boolean;
```

Comment:
Why is this necessary? Shouldn't the databaseName be non-empty if they specified the database? If the user specifies the master DB then I would say this should act like any other DB connection.

Reply:
This was mainly to fix how isMaster was determining the connection. The issue for PG was if the user specified postgres database, it would only open up to server level. Since the databaseName property ends up being set to postgres as default, wanted a way to differentiate knowing how it was set.

Reply:
What do you mean "it was set to postgres as a default"? Are you saying that the databaseName itself is updated by the code somewhere even if the user doesn't specify one in the connection dialog?

There's lots of issues with our core code with how we handle default DBs and what we do differently when a user specifies a DB - but it doesn't seem like adding this property is the direction we should be heading.